empire.go

```diff
@@ -453,8 +457,56 @@ func (opts RunOpts) Validate(e *Empire) error {
 	return e.requireMessages(opts.Message)
 }
 
+func (e *Empire) confirm(ctx context.Context, user *User, action string, resource string, params map[string]string) error {
+	if e.ActionConfirmer == nil {
+		return nil
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, time.Second*30)
+	defer cancel()
+
+	errCh := make(chan error, 1)
+	go func() {
+		var (
+			ok  bool
+			err error
+		)
+
+		ok, err = e.ActionConfirmer.Confirm(ctx, user, action, resource, params)
+		if err != nil {
+			errCh <- err
+			return
+		}
+
+		if !ok {
+			err = errors.New("not authorized")
+		}
+
+		errCh <- err
+	}()
+
+	select {
+	case err := <-errCh:
+		return err
+	case <-ctx.Done():
+		return errors.New("timed out waiting to confirm action")
+	}
+
+	return nil
+}
+
 // Run runs a one-off process for a given App and command.
 func (e *Empire) Run(ctx context.Context, opts RunOpts) error {
+	if err := e.confirm(ctx, opts.User, "Run", opts.App.Name, nil); err != nil {
```

Comment:
This ends up looking pretty [similar](https://github.com/remind101/empire/blob/e461e9bfd577acba67fb220bb1d6b77a3868da1a/empire.go#L197-L199) to how we check if an action is authorized in the ACL PR. May make sense to combine those, so that you can configure what actions require a confirmation via security policies.